### PR TITLE
Cloud error fix

### LINKED ISSUE #11985: Cloud upload crash (webdav)
* KOReader version: koreader-kindlepw2-v2024.04-147-gf33059323_2024-06-05
* Device: Kindle Scribe (5.16.2)

#### Issue
KOReader shows `Uploading file` and then crashes after about 1 minute.

#### Steps to reproduce
Connect to Wi-Fi
Go to FM
Go to Cloud Storage (Yandex.Disk via webdav)
Wait 1 min - crash to Kindle home.

##### `crash.log` (if applicable)
[cloud_upload_crash.txt](https://github.com/user-attachments/files/15597791/cloud_upload_crash.txt)

Relevant part:
```
./luajit: frontend/apps/cloudstorage/webdavapi.lua:216: attempt to compare string with number
stack traceback:
	frontend/apps/cloudstorage/webdavapi.lua:216: in function 'uploadFile'
	frontend/apps/cloudstorage/webdav.lua:55: in function 'uploadFile'
	frontend/apps/cloudstorage/cloudstorage.lua:642: in function 'action'
	frontend/ui/uimanager.lua:1020: in function '_checkTasks'
	frontend/ui/uimanager.lua:1470: in function 'handleInput'
	frontend/ui/uimanager.lua:1590: in function 'run'
	./reader.lua:265: in main chunk
	[C]: at 0x00013ebd
lipc-wait-event exited normally with status: 0
```

### PR DESCRIPTION
**Pull request by mergen3107**
Fixes #11985

It now shows InfoMessage `Could not upload file (network error)` if upload `code` is nil.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12000)

### CODE REVIEW

File: frontend/apps/cloudstorage/webdavapi.lua [hunk 1]
```diff
@@ -213,6 +213,10 @@ function WebDavApi:uploadFile(file_url, user, pass, local_path, etag)
         }
     })
     socketutil:reset_timeout()
+    if code == "timeout" or "wantread" or "sink timeout" then
```
Comment (Frenzie) [inline frontend/apps/cloudstorage/webdavapi.lua:216]: Maybe just type() ~= "number"?

Comment (hius07) [reply frontend/apps/cloudstorage/webdavapi.lua:216]: I'd add it in
https://github.com/koreader/koreader/blob/e8ed3b0681ca50c2d1c11bed768e249510b8c6dd/frontend/apps/cloudstorage/webdavapi.lua#L220

**Comment by mergen3107** [comment]
@hius07 @Frenzie 
Is this way better?

**Comment by hius07** [comment]
I think it's not very nice to duplicate the same `code` comparisons in webdav and webdavapi, but these duplications exist in other methods, so okay.

**Review by hius07 — APPROVED** [review]

**Comment by Frenzie** [comment]
> I think it's not very nice to duplicate the same `code` comparisons in webdav and webdavapi, but these duplications exist in other methods, so okay.

A very similar issue exists in #11970 btw (i.e., my proposed fix there is a half fix but I wanted to take some more time to think about the bigger problem).

**Pull request merged**